core-container/entry.sh

```diff
@@ -16,88 +16,487 @@
+    echo "EXTERNAL_HOSTNAME=${EXTERNAL_HOSTNAME}"
+
+    # Create the certificates
+    if [ -n "$EXTERNAL_HOSTNAME" ]; then
+        echo "Creating the TSL certificates"
+        # mkdir -p .kcp
+        cd .kcp
+        eval pieces_external=($(kubestellar-ensure-kcp-server-creds ${EXTERNAL_HOSTNAME}))
+        eval pieces_cluster=($(kubestellar-ensure-kcp-server-creds ${KUBESTELLAR_SERVICE})) #############
+        cd ..
+        echo "TLS certificates for server ${EXTERNAL_HOSTNAME}:"
+        echo ${pieces_external[0]}
+        echo ${pieces_external[1]}
+        echo ${pieces_external[2]}
+        echo "TLS certificates for server ${KUBESTELLAR_SERVICE}:"
+        echo ${pieces_cluster[0]}
+        echo ${pieces_cluster[1]}
+        echo ${pieces_cluster[2]}
+    fi
+
+    # Running kcp
+    if [ -n "$EXTERNAL_HOSTNAME" ]; then
+        echo "Removing existing apiserver keys... "
+        ! rm /home/kubestellar/.kcp/apiserver.* &> /dev/null # required to fix the restart
+        echo -n "Running kcp with TLS keys... "
+        kcp start --tls-sni-cert-key ${pieces_external[1]},${pieces_external[2]} --tls-sni-cert-key ${pieces_cluster[1]},${pieces_cluster[2]} & # &> kcp.log &
     else
-        echo " Not running in a cluster!"
+        echo -n "Running kcp without TLS keys... "
+        kcp start &
+    fi
+    echo Started.
+
+    # Waiting to be ready
+    echo "Waiting for ${KUBECONFIG}..."
+    while [ ! -f "${KUBECONFIG}" ]; do sleep 5; done
+    echo 'Waiting for "root:compute" workspace...'
+    until [ "$(kubectl ws root:compute 2> /dev/null)" != "" ]; do sleep 5;done
+    echo '"root:compute" workspace is ready'.
+    echo "kcp version: $(kubectl version --short 2> /dev/null | grep kcp | sed 's/.*kcp-//')"
+    kubectl ws root
+
+    # Generate the external.kubeconfig
+    if [ -n "$EXTERNAL_HOSTNAME" ] && [ ! -d "${PWD}/.kcp-${EXTERNAL_HOSTNAME}" ]; then
+        echo Creating external.kubeconfig...
+        switch-domain .kcp/admin.kubeconfig .kcp/external.kubeconfig root ${EXTERNAL_HOSTNAME} ${EXTERNAL_PORT} ${pieces_external[0]}
+        switch-domain .kcp/admin.kubeconfig .kcp/cluster.kubeconfig root ${KUBESTELLAR_SERVICE} 6443 ${pieces_cluster[0]}
+    fi
+
+    # Ensure kubeconfig secret
+    echo Creating the kubestellar secret...
+    (
+        KUBECONFIG=
+        if [ -n "${EXTERNAL_HOSTNAME}" ]; then
+            kubectl create secret generic kubestellar --from-file="${PWD}/.kcp/admin.kubeconfig" --from-file="${PWD}/.kcp/cluster.kubeconfig" --from-file="${PWD}/.kcp/external.kubeconfig"
+        else
+            kubectl create secret generic kubestellar --from-file="${PWD}/.kcp/admin.kubeconfig"
+        fi
+    )
+    echo "***READY***"
+    sleep infinity
+}
+
+
+function run_init() {
+    echo "--< Starting init >--"
+    wait_kcp_ready
+    kubestellar init --local-kcp false
+    # ensure_root_compute_configd
+    # ensure_espw
+    kubectl ws root
+    ensure_invs $ENSURE_IMW
+    ensure_wmws $ENSURE_WMW
```

Comment:
Please do not make the chart differ from the behavior of `kubestellar init`.  Either generalize `kubestellar init` to cover the expanded options or remove them from this edition of the chart and revisit in a later PR.

Reply:
I would rather keep this